src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -29,16 +29,24 @@ public class TelemetryActivity : IDisposable
         public Guid OperationId { get; }
 
         /// <summary> Singleton of NuGet telemetry service instance. </summary>
-        public static INuGetTelemetryService NuGetTelemetryService { get; set; }
+        public static INuGetTelemetryService? NuGetTelemetryService { get; set; }
 
         private TelemetryActivity(Guid parentId, TelemetryEvent telemetryEvent, Guid operationId)
         {
             if (telemetryEvent != null)
             {
+                if (telemetryEvent.Name is null)
+                {
+                    throw new ArgumentException(paramName: nameof(telemetryEvent), message: "Property 'Name' must not be null");
+                }
                 _telemetryActivity = NuGetTelemetryService?.StartActivity(telemetryEvent.Name);
             }
+            else
+            {
+                Debug.Fail("Looking at all references to the static Create methods, I don't think this code path is possible.");
```

Comment:
I didn't feel comfortable deleting the `if != null` statement, but looking at usage it should never be null. I thought maybe this `Debug.Fail` will give us confidence in a few weeks if none of us hit this debug point, so we can make the change later.

Reply:
Sounds good to me.